### PR TITLE
Add startup check for ambient transaction enlistment support

### PR DESCRIPTION
**Pull request by bording**
This ensures that the transport will fail on startup if the `TransactionScope` transport transaction mode has been selected when running against versions of the System.Data.SqlClient package that don't support ambient transaction enlistment.

The first version of the package that will support this is the 4.5.0 release, which is part of .NET Core 2.1.

FYI @timbussmann @andreasohlund @tmasternak

### CODE REVIEW

File: src/NServiceBus.SqlServer/SqlServerTransportInfrastructure.cs [hunk 1]
```diff
@@ -141,6 +141,33 @@ ExpiredMessagesPurger CreateExpiredMessagesPurger(SqlConnectionFactory connectio
...
+                {
+                    using (var scope = new TransactionScope(TransactionScopeOption.RequiresNew, transactionOptions, TransactionScopeAsyncFlowOption.Enabled))
+                    using (var connection = await connectionFactory.OpenNewConnection().ConfigureAwait(false))
+                    {
+                        scope.Complete();
+                    }
+                }
+                catch (NotSupportedException)
+                {
+                    var message = "The version of System.Data.SqlClient in use does not support enlisting SQL connections in ambient transactions, so the TransactionScope transport transaction mode cannot be used. Use `EndpointConfiguration.UseTransport<SqlServerTransport>().Transactions` to select a different transport transaction mode.";
```
Comment (andreasohlund) [inline src/NServiceBus.SqlServer/SqlServerTransportInfrastructure.cs]: Can we detect if the mode was defaulted by us or requested by the user and adjust the message accordingly?

Comment (bording) [reply src/NServiceBus.SqlServer/SqlServerTransportInfrastructure.cs]: I'm not sure what sort of different message you'd want to see there. What do you have in mind?

**Comment by andreasohlund** [comment]
Thinking that if they explicitly asked for TXScope it would be more likely
that they wanted DTC but lets call YAGNI on that

**Comment by timbussmann** [comment]
tested this PR against the SQL sample and the behavior seems to work fine. One question though: This exception is also thrown on send-only endpoints, which actually work fine with the transactionscope tx mode, as they never open a receive transaction. So theoretically we don't have to enforce a configuration downgrade on send-only endpoints.

**Pull request merged**